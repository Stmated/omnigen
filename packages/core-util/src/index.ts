
export * from './equality/index.js';
export * from './interpret/index.js';
export * from './options/index.js';
export * from './parse/index.js';
export * from './util/index.js';
export * from './visit/index.js';
export * from './write/index.js';
