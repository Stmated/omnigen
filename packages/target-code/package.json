{
  "name": "@omnigen/target-code",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "exports": {
    ".": "./src/index.ts",
    "./ast": "./src/ast/Code.ts"
  },
  "scripts": {
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
    "dev": "tsc -w",
    "lint": "TIMING=1 eslint src/**/*.ts* --fix",
    "test": "vitest run",
    "test:watch": "vitest"
  },
  "dependencies": {
    "@omnigen/api": "workspace:*",
    "@omnigen/core": "workspace:*",
    "@omnigen/core-log": "workspace:*",
    "@omnigen/core-plugin": "workspace:*",
    "@omnigen/plugin": "workspace:*",
    "zod": "^3.25.76"
  },
  "devDependencies": {
    "@omnigen/config-typescript": "workspace:*",
    "@omnigen/eslint-config-custom": "workspace:*",
    "@omnigen/utils-test": "workspace:*",
    "typescript": "^5.9.2"
  }
}
