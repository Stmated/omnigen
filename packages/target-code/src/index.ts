
export * from './options/CodeOptions';

export * from './ast/CodeAstUtils';

export * from './ast/transform/AddObjectDeclarationsCodeAstTransformer.ts';
export * from './ast/transform/AddConstructorCodeAstTransformer.ts';
export * from './ast/transform/AddAdditionalPropertiesInterfaceAstTransformer';
export * from './ast/transform/InnerTypeCompressionAstTransformer';
export * from './ast/transform/PackageResolverAstTransformer';
export * from './ast/transform/AddAccessorsForFieldsAstTransformer';
export * from './ast/transform/AddAbstractAccessorsAstTransformer';
export * from './ast/transform/AddFieldsAstTransformer';
export * from './ast/transform/AddGeneratedCommentAstTransformer';
export * from './ast/transform/AddCommentsAstTransformer';
export * from './ast/transform/ReorderMembersAstTransformer';
export * from './ast/transform/SimplifyGenericsAstTransformer';
export * from './ast/transform/ResolveGenericSourceIdentifiersAstTransformer';
export * from './ast/transform/RemoveConstantParametersAstTransformer';
export * from './ast/transform/ToConstructorBodySuperCallAstTransformer';
export * from './ast/transform/RemoveEnumFieldsCodeAstTransformer.ts';
export * from './ast/transform/MethodToGetterCodeAstTransformer.ts';
export * from './ast/transform/AddCompositionMembersCodeAstTransformer.ts';

export * from './visitor/CodeVisitor';
export * from './visitor/FreeTextVisitor';

export * from './util/FreeTextUtils';
export * from './util/CodeUtil';

export * from './render/CodeRenderer';

export * from './reduce/CodeAstReducer';

export * from './parse/transform/DeleteUnnecessaryCompositionsJavaModelTransformer';
export * from './parse/transform/InterfaceExtractorModelTransformer.ts';

export * as Code from './ast/Code';
