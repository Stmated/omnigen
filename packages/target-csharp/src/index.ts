export * from './options';
export * from './visit';
export * from './features';
export * from './parse';
export * from './ast';

export * as CSharpPlugins from './CSharpPluginInit';
