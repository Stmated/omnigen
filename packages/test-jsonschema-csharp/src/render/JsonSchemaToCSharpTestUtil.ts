import {RenderedCompilationUnit, TargetOptions} from '@omnigen/api';
import {PluginManager} from '@omnigen/plugin';
import {CSharpOptions, CSharpPlugins} from '@omnigen/target-csharp';
import {LoggerFactory} from '@omnigen/core-log';
import {JsonSchemaPlugins} from '@omnigen/parser-jsonschema';
import {FileContext} from '@omnigen/core-plugin';

const logger = LoggerFactory.create(import.meta.url);

export class JsonSchemaToCSharpTestUtil {

  public static async render(inPath: string, options?: Partial<CSharpOptions & TargetOptions>): Promise<RenderedCompilationUnit[]> {

    const pm = new PluginManager({includeAuto: true});

    logger.info(`Running plugin manager with ${JsonSchemaPlugins.JsonSchemaPlugin.name} and ${CSharpPlugins.CSharpPlugin.name}`);

    const ctx: FileContext = {
      file: inPath,
      arguments: {
        target: 'csharp',
        ...options,
      },
    };

    const result = await pm.execute({
      ctx: ctx,
      stopAt: CSharpPlugins.CSharpRendererCtxOut,
    });

    return result.result.ctx.compilationUnits;
  }
}
