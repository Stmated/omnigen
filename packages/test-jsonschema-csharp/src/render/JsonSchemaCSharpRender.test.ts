import {describe, expect, test, vi} from 'vitest';
import {JsonSchemaToCSharpTestUtil} from './JsonSchemaToCSharpTestUtil.ts';
import {Util} from '@omnigen/core-util';
import {SerializationLibrary} from '@omnigen/target-csharp';
import {IncludeExampleCommentsMode, PropertyTypeCommentMode} from '@omnigen/target-code';

describe('jsonschema-csharp-render', () => {

  test('string_union', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(Util.getPathFromRoot('./packages/parser-jsonschema/examples/string_union.json'));
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu).toHaveLength(1);
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('x-enum-varnames', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(Util.getPathFromRoot('./packages/parser-jsonschema/examples/keep_x_enum_varnames.json'));
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('decorated_types', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(
      Util.getPathFromRoot('./packages/parser-jsonschema/examples/decorated_types.json'),
      {
        compressSoloReferencedTypes: false,
      },
    );
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('enum_string_composition', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(Util.getPathFromRoot('./packages/parser-jsonschema/examples/enum_string_composition.json'), {
      compressSoloReferencedTypes: false,
      typeCommentsOnProperties: PropertyTypeCommentMode.ALWAYS,
    });
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('interface_order', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(Util.getPathFromRoot('./packages/parser-jsonschema/examples/interface_order.json'), {
      compressSoloReferencedTypes: false,
      serializationLibrary: SerializationLibrary.NONE,
    });
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('inline_boolean', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(Util.getPathFromRoot('./packages/parser-jsonschema/examples/inline_boolean.json'), {
      serializationLibrary: SerializationLibrary.NONE,
      includeExampleCommentsMode: IncludeExampleCommentsMode.SKIP,
      compressSoloReferencedTypes: true,
      compressUnreferencedSubTypes: true,
      singleFile: false,
    });
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('class_union', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(Util.getPathFromRoot('./packages/parser-jsonschema/examples/class_union.json'), {
      compressSoloReferencedTypes: false,
      compressUnreferencedSubTypes: false,
      serializationLibrary: SerializationLibrary.NEWTONSOFT,
      includeExampleCommentsMode: IncludeExampleCommentsMode.SKIP,
    });
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('enum_inheritance', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(Util.getPathFromRoot('./packages/parser-jsonschema/examples/enum_inheritance.json'));
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('one_of_same_type', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(Util.getPathFromRoot('./packages/parser-jsonschema/examples/one_of_same_type.json'));
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('discriminator', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToCSharpTestUtil.render(Util.getPathFromRoot('./packages/parser-jsonschema/examples/discriminator.json'));
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      expect(cu[0].content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });
});
