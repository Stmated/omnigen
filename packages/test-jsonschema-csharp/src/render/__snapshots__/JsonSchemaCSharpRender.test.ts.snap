// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`jsonschema-csharp-render > class_union 1`] = `
[
  "AbortData.cs",
  "Data.cs",
  "DefaultData.cs",
  "StringBoolean.cs",
]
`;

exports[`jsonschema-csharp-render > decorated_types 1`] = `
[
  "Order.cs",
  "Shop.cs",
]
`;

exports[`jsonschema-csharp-render > discriminator 1`] = `
[
  "ResponseData.cs",
]
`;

exports[`jsonschema-csharp-render > enum_inheritance 1`] = `
[
  "Kind1.cs",
]
`;

exports[`jsonschema-csharp-render > enum_string_composition 1`] = `
[
  "DataEntry.cs",
  "TransferState.cs",
]
`;

exports[`jsonschema-csharp-render > inline_boolean 1`] = `
[
  "AccountNotificationData.cs",
]
`;

exports[`jsonschema-csharp-render > interface_order 1`] = `
[
  "AbstractRequestData.cs",
  "BalanceRequestData.cs",
  "CancelChargeRequestData.cs",
  "CancelChargeRequestDataBase.cs",
  "ICancelChargeRequestDataBase.cs",
]
`;

exports[`jsonschema-csharp-render > one_of_same_type 1`] = `
[
  "RequestData.cs",
]
`;

exports[`jsonschema-csharp-render > string_union 1`] = `
[
  "Container.cs",
]
`;

exports[`jsonschema-csharp-render > x-enum-varnames 1`] = `
[
  "SomeEnum.cs",
]
`;
